psij-ci-setup

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+FORCE=0
+
+if [ "$1" == "-f" ]; then
+    FORCE=1
+    shift
+fi
+
+MYPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+echo
+echo "================================================================"
+echo "This script will install requirements for the PSI/J CI tests and"
+echo "add a cron job to run the tests once a day at some random time. "
+echo "================================================================"
+echo
+
+RESPONSE=""
+
+while [ "$RESPONSE" != "C" ] && [ "$RESPONSE" != "X" ]; do
+
+    echo -n "Would you like to (C)ontinue or E(x)it? "
+    read -n1 RESPONSE
+    echo
+    RESPONSE=${RESPONSE^}
+    
+    if [ "$RESPONSE" == "X" ]; then
+        echo "Operation canceled"
+        exit 1
+    fi
+done
+
+cd "$MYPATH"
+
+pip install --user -r requirements-dev.txt
```

Comment:
Installing in `--user` is not a good idea for shared file systems which are used on resources with different architectures.  Have been there, done that, and lost a week debugging why the heck some module did not work... :-P  Also, that behavior would be surprising if the script runs in a virtual or conda env I think?

Reply:
I'm assuming that the main scenario where we would want this to function is resources that the user does not have root access to. I'd be happy to hear alternatives, but system installs don't seem to be in the realm of plausible here.

Reply:
If the user is in a virtual/conda environment, though, they wouldn't need ``root``. I agree with Andre that ``--user`` could be confusing if you find that the packages actually weren't installed to your venv. And I don't think it's unreasonable to ask that anyone running the tests be in a venv.

If it's necessary we could always try a regular install and then on failure emit a warning and fall back to a user install.

Reply:
> I don't think it's unreasonable to ask that anyone running the tests be in a venv.

I worry that requiring things will tend to deter people from running the tests.

Perhaps it is wiser to attempt to detect if inside a virtual/conda env and disable "--user" for those cases.

Reply:
One of the difficulties I can see with venv and conda is that running things from cron may not be that straightforward, since the relevant environment may not be activated. Thoughts?

Reply:
That is a good point about cron. Is there maybe a way to associate the scripts with a venv, something like "I want this script to run every TIME in venv PATH"?

Reply:
So I wrote a quick update to the scripts to detect if running inside conda or venv, save the venv path or conda env name, and issue the relevant activate command before invoking the tests. I want to refine and test it a bit before committing, but it should take care of the issue.

In principle the idea is that you should be able to run the setup script straight up, under a venv, or under conda, and it will do the right thing to restore the respective environment when running the tests. We'll have to see how well it works in practice, but it should tick all the boxes.